packages/core/src/readers/SimpleDirectoryReader.edge.ts

```diff
@@ -58,7 +65,37 @@ export class SimpleDirectoryReader implements BaseReader {
     }
 
     const docs: Document[] = [];
+    const filePathQueue: string[] = [];
+
     for await (const filePath of walk(fs, directoryPath)) {
+      filePathQueue.push(filePath);
+    }
+
+    const workerPromises = Array.from({ length: numWorkers }, () =>
+      this.processFiles(filePathQueue, fs, defaultReader, fileExtToReader),
+    );
+
+    const results = await Promise.all(workerPromises);
+    docs.push(...results.flat());
```

Comment:
```suggestion
    docs.push(...results.flat());
```

use `Array.prototype.concat`, push will throw error when results larger than 1000 items